src/cli/commands/install.js

```diff
@@ -431,6 +431,10 @@ export class Install {
       this.scripts.setArtifacts(artifacts);
     }
 
+    if (Object.getPrototypeOf(this) === Install.prototype && depRequests.length === 0) {
```

Comment:
I think you can just enhance the code there, you probably don't need to check for Install.prototype.
The logic should be "don't save lockfile if there are no dependencies"